drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -348,21 +359,28 @@ int stm32_clock_control_init(const struct device *dev)
 {
 	LL_UTILS_ClkInitTypeDef s_ClkInitStruct;
 	uint32_t hclk_prescaler;
+	uint32_t flash_prescaler;
 #if STM32_SYSCLK_SRC_HSE || STM32_SYSCLK_SRC_MSI
-	uint32_t old_hclk_freq;
 	uint32_t new_hclk_freq;
+	uint32_t old_flash_freq;
+	uint32_t new_flash_freq;
 #endif
 
 	ARG_UNUSED(dev);
 
 	/* configure clock for AHB/APB buses */
 	config_bus_clk_init((LL_UTILS_ClkInitTypeDef *)&s_ClkInitStruct);
 
-	/* update local hclk prescaler variable */
-#if defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_SOC_SERIES_STM32WLX)
+	/* update local hclk and flash-clk prescaler variable */
+#if defined(CONFIG_SOC_SERIES_STM32WBX)
 	hclk_prescaler = s_ClkInitStruct.CPU1CLKDivider;
+	flash_prescaler = s_ClkInitStruct.AHB4CLKDivider;
+#elif defined(CONFIG_SOC_SERIES_STM32WLX)
+	hclk_prescaler = s_ClkInitStruct.CPU1CLKDivider;
+	flash_prescaler = s_ClkInitStruct.AHB3CLKDivider;
 #else
 	hclk_prescaler = s_ClkInitStruct.AHBCLKDivider;
+	flash_prescaler = s_ClkInitStruct.AHBCLKDivider;
```

Comment:
Nit: 
For readability purpose, I'd do:
flash_prescaler = hclk_prescaler;
Just a suggestion, not blocking.

Reply:
Done.